go-controller/pkg/config/config.go

```diff
@@ -1109,6 +1121,21 @@ func buildGatewayConfig(ctx *cli.Context, cli, file *config) error {
 	if Gateway.Mode != GatewayModeShared && Gateway.VLANID != 0 {
 		return fmt.Errorf("gateway VLAN ID option: %d is supported only in shared gateway mode", Gateway.VLANID)
 	}
+
+	// Validate v4 and v6 join subnets
+	_, v4JoinCIDR, err := net.ParseCIDR(Gateway.V4JoinSubnet)
+	if err != nil {
```

Comment:
Do we need the same check as on L1132? I.e: `if err != nil || utilnet.IsIPv6CIDRString(Gateway.v4JoinCIDR)`

Reply:
The `ParseCIDR` takes care of verifying it's a valid v4/v6 subnet. the `IsIPv6CIDRString` is an additional check to make sure that it indeed is a v6 address. I will add an explicit check for verifying that the `V4JoinSubnet` is a v4 cidr.